akka-http/src/main/scala/akka/http/scaladsl/server/directives/FileAndResourceDirectives.scala

```diff
@@ -222,7 +222,14 @@ object FileAndResourceDirectives extends FileAndResourceDirectives {
             ResourceFile(url, entry.getSize, entry.getTime)
           }
         } finally jar.close()
-      case _ ⇒ None
+      case _ ⇒
+        val conn = url.openConnection()
```

Comment:
You probably want to do a `url.setConnectTimeout` and `url.setReadTimeout`, since the default is possibly infinite.

And maybe have a unit test for this?